### PR TITLE
Update modbus switch status verify

### PR DESCRIPTION
**Pull request by crug80**
## Proposed change

Updates reflecting [PR #127444](https://github.com/home-assistant/core/pull/127444)

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [X] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/127444
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Enhanced documentation for Modbus integration, including updates to configuration parameters for switches, sensors, covers, and climates.
	- Introduced a parameters usage matrix for better configuration guidance.
- **Bug Fixes**
	- Clarified descriptions for existing parameters to improve user understanding across various integrations.
- **Documentation**
	- Improved structure and examples throughout the Modbus integration documentation for better usability.
	- Updated documentation to reflect changes in `state_on` and `state_off` attributes for switches, allowing both single integers and lists of integers.

### CODE REVIEW

File: source/_integrations/modbus.markdown [hunk 1]
```diff
@@ -1516,15 +1516,15 @@ switches:
             input:
               description: "Input register."
         state_on:
-          description: "Value when switch is on."
+          description: "Value(s) when switch is on."
           required: false
           default: "Same as `command_on`"
-          type: integer
+          type: integer|list
```
Comment (MartinHjelmare) [inline source/_integrations/modbus.markdown]: I don't think it's possible to define a union of different types like this. Only one type per key is supported as far as I know.

Comment (crug80) [reply source/_integrations/modbus.markdown]: The type could be an integer or a list of integers (see related PR to the codebase). I used the same mode used in the modbus documentation to explain similar cases. In example: `address integer | list `
Probably I should add a space between..

Comment (crug80) [reply source/_integrations/modbus.markdown]: Ehm.. something unwanted happened.. Probably my bad.

Comment (jbouwh) [reply source/_integrations/modbus.markdown]: @crug80 I have rebased the PR, locally you should do a `git pull`

File: source/_integrations/modbus.markdown [hunk 2]
```diff
@@ -1516,15 +1516,15 @@ switches:
             input:
               description: "Input register."
         state_on:
-          description: "Value when switch is on."
+          description: "Value(s) when switch is on."
           required: false
           default: "Same as `command_on`"
-          type: integer
+          type: integer | list
```
Comment (jbouwh) [inline source/_integrations/modbus.markdown]:
```suggestion
          type: integer
```

File: source/_integrations/modbus.markdown [hunk 3]
```diff
@@ -1516,15 +1516,15 @@ switches:
...
         state_off:
-          description: "Value when switch is off."
+          description: "Value(s) when switch is off."
           required: false
           default: "Same as `command_off`"
-          type: integer
+          type: integer | list
```
Comment (jbouwh) [inline source/_integrations/modbus.markdown]:
```suggestion
          type: integer
```

**Review by jbouwh — COMMENTED** [review]
The union will not be accepted by the linters.

File: source/_integrations/modbus.markdown [hunk 4]
```diff
@@ -1516,15 +1516,15 @@ switches:
             input:
               description: "Input register."
         state_on:
-          description: "Value when switch is on."
+          description: "Value(s) when switch is on."
```
Comment (jbouwh) [inline source/_integrations/modbus.markdown]:
```suggestion
          description: "Value(s) when switch is on. The value must be an `integer` or a list of integers."
```

File: source/_integrations/modbus.markdown [hunk 5]
```diff
@@ -1516,15 +1516,15 @@ switches:
...
           required: false
           default: "Same as `command_on`"
-          type: integer
+          type: integer | list
         state_off:
-          description: "Value when switch is off."
+          description: "Value(s) when switch is off."
```
Comment (jbouwh) [inline source/_integrations/modbus.markdown]:
```suggestion
          description: "Value(s) when switch is off.  The value must be an `integer` or a list of integers."
```

File: source/_integrations/modbus.markdown (hunk 2 above)
Comment (frenck) [reply source/_integrations/modbus.markdown]:
```suggestion
          type: [integer, list]
```

File: source/_integrations/modbus.markdown (hunk 3 above)
Comment (frenck) [reply source/_integrations/modbus.markdown]:
```suggestion
          type: [integer, list]
```

**Review by frenck — APPROVED** [review]
Thanks, @crug80 👍 

../Frenck

**Pull request merged**